src/libs/actions/PersonalDetails.js

```diff
@@ -87,6 +88,11 @@ function formatPersonalDetails(personalDetailsList) {
         const pronouns = lodashGet(personalDetailsResponse, 'pronouns', '');
         const timezone = lodashGet(personalDetailsResponse, 'timeZone', CONST.DEFAULT_TIME_ZONE);
 
+        // Update the users timezone if they have it set to automatic
+        if (login === currentUserEmail && _.isObject(timezone) && timezone.automatic) {
```

Comment:
This is not really updating the NVP, IMO it should update the NVP. Also, it feels a bit weird that we are doing this logic here... we will be guessing the timezone all the time we format your details? Doesn't it make more sense to do this once on app init?

Reply:
Oh, yeah good point, I should make sure I sent the NVP as well.

Reply:
As for doing it all the time, we only call `personalDetails` fetch in Authscreens.js in two places. Once on init, and then every 30 min to update the details as necessary. Since we want this to be automatic, I don't see an issue with it running every 30 min, but maybe I am too lenient.

Reply:
I don't know, feels weird to me that we do this here... you call `formatPersonalDetails` and all of a sudden an NVP is set... it's weird.

Reply:
Yeah, I can understand that - you wouldn't expect a function called `formatXYZ` to set any values. I could move this logic into it's own function and call it in `fetch()` which isn't much better, but makes a little more sense since the value from the DB might be outdated if the user has moved to a new timezone.

Reply:
Can we update this on app start instead? Somewhere around [here](https://github.com/Expensify/Expensify.cash/blob/3ac2f7aa3e0014c2fc7f3ca012486eff69f19bbe/src/libs/Navigation/AppNavigator/AuthScreens.js#L74-L110) maybe?

Reply:
That's where I had it before, but there were issues because we don't necessarily have the personalDetails at that point. I suppose we could make `PersonalDetails.fetch()` return a promise and when that resolves run the logic for updating the timezone if it's set to automatic.

Reply:
I think we should not use the personal details at all there, just get the NVP from your own account.

Reply:
Hmm, I think that'll potentially cause some issues since we are already fetching personalDetails (which contains the timezone) here. We'd need to be sure that the `myPersonalDetails` and `personalDetails` onyx keys are updated correctly after we modify the timezone. 

Based on your suggestion, I think we'd end up doing something like this in `AuthScreens.js` (adding some small modifications to `NameValuePairs.get` as well).  

```JS
NameValuePair.get(CONST.NVP.TIMEZONE, CONST.DEFAULT_TIMEZONE)
    .then((value) => {
        const timezone = value;
        if (timezone.automatic) {
            timezone.selected = moment.tz.guess(true);
            NameValuePair.set(CONST.NVP.TIMEZONE, timezone);
            PersonalDetails.fetch();
        }
    });
```

Is that along the lines of what you were thinking?

Reply:
No, not really since actions should not return a promise, you need to subscribe to the onyx key, no?

Reply:
Ok, chatted 1:1 and I messed around with this a bit. Works now!